File: src/main/java/frc/robot/util/NavX.java
package frc.robot.util;

import com.kauailabs.navx.frc.AHRS;
import edu.wpi.first.math.MathUtil;
import edu.wpi.first.math.geometry.Rotation2d;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

/**
 * Used to find the angle of the robot in its current position
 * to keep orientated
 */
public class NavX extends AHRS {

    private double angleOffset = 0.0;

    @Override 
    public double getAngle() {
        double angle = super.getAngle() - angleOffset;
        angle = MathUtil.inputModulus(angle, -180.0, 180.0);
        return angle;
    }

    public void setAngleOffset(double setValue) {
        this.angleOffset = setValue + super.getAngle();
    }
}

File: src/main/java/frc/robot/util/NavX.kt
package frc.robot.util

import com.kauailabs.navx.frc.AHRS
import edu.wpi.first.math.MathUtil
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard

/**
 * Used to find the angle of the robot in its current position
 * to keep orientated
 */
class NavX : AHRS() {

    private var angleDelta = 0.0

    init { angleDelta = angle }

    override fun getAngle(): Double {
        var angle = super.getAngle() - angleDelta
        angle = MathUtil.inputModulus(angle, -180.0, 180.0)

        return angle
    }

    override fun reset() {
        angleDelta = super.getAngle()
//        if (isCalibrating) {
//            println("INFO: The imu used OUR program to reset")
//            angleDelta = super.getAngle()
//        } else {
//            println("INFO: The imu used the super reset")
//            super.reset()
//        }
    }

}